lib/web_ui/lib/src/engine/semantics/semantics_helper.dart

```diff
@@ -0,0 +1,316 @@
+    return _enableSemantics.prepareAccesibilityPlaceholder();
+  }
+}
+
+abstract class _EnableSemantics {
+  /// A temporary placeholder used to capture a request to activate semantics.
+  html.Element _semanticsPlaceholder;
+
+  /// We do not immediately enable semantics when the user requests it, but
+  /// instead wait for a short period of time before doing it. This is because
+  /// the request comes as a tap on the [_semanticsPlaceholder]. The tap,
+  /// depending on the browser, comes as a burst of events. For example, Safari
+  /// sends "touchstart", "touchend", and "click". So during a short time period
+  /// we consume all events and prevent forwarding to the framework. Otherwise,
+  /// the events will be interpreted twice, once as a request to activate
+  /// semantics, and a second time by Flutter's gesture recognizers.
+  @visibleForTesting
+  Timer semanticsActivationTimer;
+
+  /// The number of events we processed that could potentially activate
+  /// semantics.
+  int semanticsActivationAttempts = 0;
+
+  /// Instructs [_tryEnableSemantics] to remove [_semanticsPlaceholder].
+  ///
+  /// On Chrome the placeholder is removed upon any next event.
+  ///
+  /// On Safari the placeholder is removed upon the next "touchend" event. This
+  /// is to prevent Safari from swallowing the event that happens on an element
+  /// that's being removed. Chrome doesn't have this issue.
+  bool _schedulePlaceholderRemoval = false;
+
+  /// Whether we are waiting for the user to enable semantics.
+  bool get _isWaitingToEnableSemantics => _semanticsPlaceholder != null;
+
+  bool shouldEnableSemantics(html.Event event) {
+    if (!_isWaitingToEnableSemantics) {
+      // Forward to framework as normal.
+      return true;
+    } else {
+      return tryEnableSemantics(event);
+    }
+  }
+
+  /// Attempts to activate semantics.
+  ///
+  /// Returns true if the `event` is not related to semantics activation and
+  /// should be forwarded to the framework.
+  bool tryEnableSemantics(html.Event event);
+
+  /// Creates the placeholder for accesibility.
+  ///
+  /// Puts it inside the glasspane.
+  ///
+  /// On focus the element announces that accessibility can be enabled by
+  /// tapping/clicking. (Announcement depends on the assistive technology)
+  html.Element prepareAccesibilityPlaceholder();
+}
+
+@visibleForTesting
+class DesktopEnableSemantics extends _EnableSemantics {
+  @override
+  bool tryEnableSemantics(html.Event event) {
+    if (_schedulePlaceholderRemoval) {
+      _semanticsPlaceholder.remove();
+      _semanticsPlaceholder = null;
+      semanticsActivationTimer = null;
+      return true;
+    }
+
+    if (EngineSemanticsOwner.instance.semanticsEnabled) {
+      // Semantics already enabled, forward to framework as normal.
+      return true;
+    }
+
+    // In touch screen laptops, the touch is received as a mouse click
+    const List<String> kInterestingEventTypes = <String>[
```

Comment:
For contains only, use a Set instead of a List - here and elsewhere

Reply:
done.